examples~/quickstart/client/Program.cs

```diff
@@ -88,44 +96,59 @@ void PrintMessage(Message message)
     Console.WriteLine($"{senderName}: {message.Text}");
 }
 
-void Message_OnInsert(Message insertedValue, ReducerEvent? dbEvent)
+void Message_OnInsert(EventContext ctx, Message insertedValue)
 {
-    if (dbEvent != null)
+    if (ctx.Reducer is not Event<Reducer>.SubscribeApplied)
     {
         PrintMessage(insertedValue);
     }
 }
 
-void Reducer_OnSetNameEvent(ReducerEvent reducerEvent, string name)
+void Reducer_OnSetNameEvent(EventContext ctx, string name)
 {
-    if (reducerEvent.Identity == local_identity && reducerEvent.Status is UpdateStatus.Failed)
+    if (ctx.Reducer is Event<Reducer>.Reducer reducer)
     {
-        Console.Write($"Failed to change name to {name}");
+        var e = reducer.ReducerEvent;
+        if (e.CallerIdentity == local_identity && e.Status is Status.Failed(var error))
+        {
+            Console.Write($"Failed to change name to {name}: {error}");
+        }
     }
 }
 
-void Reducer_OnSendMessageEvent(ReducerEvent reducerEvent, string text)
+void Reducer_OnSendMessageEvent(EventContext ctx, string text)
 {
-    if (reducerEvent.Identity == local_identity && reducerEvent.Status is UpdateStatus.Failed)
+    if (ctx.Reducer is Event<Reducer>.Reducer reducer)
     {
-        Console.Write($"Failed to send message {text}");
+        var e = reducer.ReducerEvent;
+        if (e.CallerIdentity == local_identity && e.Status is Status.Failed(var error))
+        {
+            Console.Write($"Failed to send message {text}: {error}");
+        }
     }
 }
 
-void OnConnect()
+void OnConnect(Identity identity, string authToken)
 {
-    SpacetimeDBClient.instance.Subscribe(new List<string> { "SELECT * FROM User", "SELECT * FROM Message" });
+    local_identity = identity;
+    AuthToken.SaveToken(authToken);
+
+    conn!.Subscribe(new List<string> { "SELECT * FROM User", "SELECT * FROM Message" });
 }
 
-void OnIdentityReceived(string authToken, Identity identity, Address _address)
+void OnConnectError(WebSocketError? error, string message)
 {
-    local_identity = identity;
-    AuthToken.SaveToken(authToken);
+
+}
+
+void OnDisconnect(DbConnection conn, WebSocketCloseStatus? status, WebSocketError? error)
```

Comment:
The proposal has only one error object - do we need both args here and in OnConnectError?

Reply:
Ah hmm, I think its meant to follow the idiomatic error handling in this case.

Reply:
Looking at the invocations in WebSocket.cs, there's information lost if we remove that parameter; its better to update the proposal to reflect the implementation.